cli/commands/play.go

```diff
@@ -95,6 +97,17 @@ var run = func(cmd *cobra.Command, args []string) {
 
 	ruleset := getRuleset(Seed, snakeStates)
 	state := initializeBoardFromArgs(ruleset, snakeStates)
+	exportGame := Output != ""
+
+	var gameExporter GameExporter
+	if exportGame {
+		gameExporter = GameExporter{
+			game:          createClientGame(ruleset),
+			snakeRequests: make([]client.SnakeRequest, 0),
+			winner:        SnakeState{},
+			isDraw:        false,
+		}
+	}
```

Comment:
Since the GameExporter isn't doing much work until the final call to `FlushToFile`, it might simplify things a bit to just always initialize it here, and remove all the `if exportGame` conditionals before the last one.

Reply:
That's a good point - I was assuming that memory usage would be minimal with short games, but we can't say for sure people won't use this for really long games with a lot of elements on the board.
Doesn't seem worth it to remove a few lines then.

Reply:
I'll add a note explaining the reasoning for future reference :smile: